library/ceph_key.py

```diff
@@ -0,0 +1,422 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ceph_key
+
+author: Sebastien Han <seb@redhat.com>
+
+short_description: Manage Cephx key(s)
+
+version_added: "2.6"
+
+description:
+    - Manage CephX creation, deletion and updates.
+    It can also list and get information about key(s)
+options:
+    cluster:
+        description:
+            - The ceph cluster name.
+        required: false
+        default: ceph
+    name:
+        description:
+            - name of the CephX key
+        required: true
+    state:
+        description:
+            - present|absent|list|update|info
+        required: true
+        choices: ['present', 'absent', 'list', 'update', 'info']
+        default: list
+    caps:
+        description:
+            - CephX key capabilities
+        required: false
+    secret:
+        description:
+            - key's secret value
+        required: false
+        default: None
+    containerized:
+        description:
+            - Weither or not this is a containerized cluster. The value is
+            assigned or not depending on how the playbook runs.
+        required: false
+        default: None
+'''
+
+EXAMPLES = '''
+- name: create cephx key
+  ceph_key:
+    name: "my_key"
+    state: present
+    caps: "{{ caps }}"
+
+- name: update cephx key
+  ceph_key:
+    name: "my_key"
+    state: update
+    caps: "{{ caps }}"
+
+- name: delete cephx key
+  ceph_key:
+    name: "my_key"
+    state: absent
+
+- name: info cephx key
+  ceph_key:
+    name: "my_key""
+    state: info
+
+- name: list cephx keys
+  ceph_key:
+    state: list
+'''
+
+RETURN = '''#  '''
+
+from ansible.module_utils.basic import AnsibleModule
+import datetime
+import os
+import struct
+import time
+import base64
+
+
+def fatal(message, module):
+    '''
+    Report a fatal error and exit
+    '''
+
+    if module:
+        module.fail_json(msg=message, rc=1)
+    else:
+        raise(Exception(message))
+
+
+def key_mode(file_path):
+    '''
+    Change mode file for a CephX key
+    Problem, how do to do on containerized deployment?
+    '''
+    os.chmod(file_path)
+
+
+def generate_secret():
+    '''
+    Generate a CephX secret
+    '''
+
+    key = os.urandom(16)
+    header = struct.pack('<hiih', 1, int(time.time()), 0, len(key))
+    secret = base64.b64encode(header + key)
+
+    return secret
+
+
+def generate_caps(cmd, type, caps):
```

Comment:
using `type` as an argument here means that it is overriding Python's builtin `type`. You could use an underscore here or a different name

Reply:
Will do, thanks.